runners/direct-java/src/main/java/org/apache/beam/runners/direct/CopyOnAccessInMemoryStateInternals.java

```diff
@@ -355,6 +379,12 @@ private boolean containedInUnderlying(StateNamespace namespace, StateTag<? super
           public <KeyT, ValueT> MapState<KeyT, ValueT> bindMap(
               StateTag<? super K, MapState<KeyT, ValueT>> address,
               Coder<KeyT> mapKeyCoder, Coder<ValueT> mapValueCoder) {
+            if (mapKeyCoder == null || mapValueCoder == null) {
```

Comment:
I think that these parameters should not be `@Nullable` but instead the spec/tag should own this exception and throw it when it discovers that it has to provide these parameters but cannot.